src/core/network/qgsnewsfeedparser.cpp

```diff
@@ -190,33 +190,73 @@ void QgsNewsFeedParser::onFetch( const QString &content )
   const QVariant json = QgsJsonUtils::parseJson( content );
 
   const QVariantList entries = json.toList();
-  QList< QgsNewsFeedParser::Entry > newEntries;
-  newEntries.reserve( entries.size() );
+  QList< QgsNewsFeedParser::Entry > fetchedEntries;
+  fetchedEntries.reserve( entries.size() );
   for ( const QVariant &e : entries )
   {
-    Entry newEntry;
+    Entry incomingEntry;
     const QVariantMap entryMap = e.toMap();
-    newEntry.key = entryMap.value( QStringLiteral( "pk" ) ).toInt();
-    newEntry.title = entryMap.value( QStringLiteral( "title" ) ).toString();
-    newEntry.imageUrl = entryMap.value( QStringLiteral( "image" ) ).toString();
-    newEntry.content = entryMap.value( QStringLiteral( "content" ) ).toString();
-    newEntry.link = entryMap.value( QStringLiteral( "url" ) ).toString();
-    newEntry.sticky = entryMap.value( QStringLiteral( "sticky" ) ).toBool();
+    incomingEntry.key = entryMap.value( QStringLiteral( "pk" ) ).toInt();
+    incomingEntry.title = entryMap.value( QStringLiteral( "title" ) ).toString();
+    incomingEntry.imageUrl = entryMap.value( QStringLiteral( "image" ) ).toString();
+    incomingEntry.content = entryMap.value( QStringLiteral( "content" ) ).toString();
+    incomingEntry.link = entryMap.value( QStringLiteral( "url" ) ).toString();
+    incomingEntry.sticky = entryMap.value( QStringLiteral( "sticky" ) ).toBool();
     bool ok = false;
-    const uint expiry = entryMap.value( QStringLiteral( "publish_to" ) ).toUInt( &ok );
+    const qlonglong expiry = entryMap.value( QStringLiteral( "publish_to" ) ).toLongLong( &ok );
     if ( ok )
-      newEntry.expiry.setSecsSinceEpoch( expiry );
-    newEntries.append( newEntry );
+      incomingEntry.expiry.setSecsSinceEpoch( expiry );
 
-    if ( !newEntry.imageUrl.isEmpty() )
-      fetchImageForEntry( newEntry );
+    fetchedEntries.append( incomingEntry );
+
+    // We also need to handle the case of modified/expired entries
+    const auto entryIter { std::find_if( mEntries.begin(), mEntries.end(), [incomingEntry]( const QgsNewsFeedParser::Entry & candidate )
+    {
+      return candidate.key == incomingEntry.key;
+    } )};
+    const bool entryExists { entryIter != mEntries.end() };
+
+    // case 1: existing entry is now expired, dismiss
+    if ( ok && expiry < mFetchStartTime )
```

Comment:
Can we rename `ok` here to something more explicit, eg `hasExpiry`? I'm concerned it might get re-used for some other purpose in future if we leave the generic name.